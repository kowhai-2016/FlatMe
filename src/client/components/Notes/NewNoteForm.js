import { Control, Errors, Form } from 'react-redux-form'
import { Modal, Button } from 'react-bootstrap'
import React from 'react'

const NewNoteForm = props => {
  return (
    <Form model='forms.newNote' onSubmit={props.onSubmit}>
      <Modal.Header closeButton>
        <Modal.Title>New note:</Modal.Title>
      </Modal.Header>
      <Modal.Body>
        <div className='newNoteForm'>
          <Control.textarea className='newNoteField' model='.content' />
        </div>
        <Errors model='forms.newNote' />
      </Modal.Body>
      <Modal.Footer>
        <Button type='submit'>Create</Button>
      </Modal.Footer>
    </Form>
  )
}

export default NewNoteForm
