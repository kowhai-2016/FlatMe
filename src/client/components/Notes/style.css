@import url('http://fonts.googleapis.com/css?family=Bad+Script');

.NoteTitle {
  color: lightseagreen;
}

.NotesContainer {
  display: flex;
  flex-wrap: wrap;
  position: relative;
  font-family: 'Bad Script', Helvetica, Sans-serif;
}

.NoteFull {
  width: 45%;
  margin-right: 20px;
  margin-top: 20px;
  background-color: #e2fffa;
  box-shadow: 1px 4px 2px;
  position: relative;
}

.NoteContent{
  padding: 10px;
  font-size: 2em;
}

.NoteAuthor {
  padding: 10px;
  text-align: right;
  padding-top: 25px;
  font-style: italic;
  font-size: 1.5em;
}

.NoteDelete {
  background-color: transparent;
  outline: none;
  border: none;
  padding-top: 5px;
}

.NoteAdd {
  font-size: 1.3em;
}

@media (max-width: 1085px) and (min-width: 992px) {
  .NoteFull {
    width: 100%;
  }
}

@media (max-width: 500px) {
  .NoteFull {
    width: 100%;
  }
}
