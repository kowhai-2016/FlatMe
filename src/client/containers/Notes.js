import { connect } from 'react-redux'

import { fetchNotes, deleteNote, addNote } from '../actions'
import Notes from '../components/Notes'

const mapStateToProps = (state, ownProps) => {
  return {
    flat: Number(ownProps.params.id),
    notes: state.notes,
    user: state.account.user,
    newNote: state.forms.newNote.content
  }
}

const mapDispatchToProps = dispatch => {
  return {
    fetchNotes: (flat) => { return dispatch(fetchNotes(flat)) },
    deleteNote: (id) => { return dispatch(deleteNote(id)) },
    addNote: (note) => { return dispatch(addNote(note)) }
  }
}

export default connect(mapStateToProps, mapDispatchToProps)(Notes)
