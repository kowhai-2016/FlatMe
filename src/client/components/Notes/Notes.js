import React from 'react'
import { Modal, Button, Col } from 'react-bootstrap'

import './style.css'

import NewNoteForm from './NewNoteForm'

export default React.createClass({
  // gets note data from api
  componentDidMount () {
    this.props.fetchNotes(this.props.flat)
  },
  // functions for opening and closing newNoteForm modal
  getInitialState () {
    return {show: false}
  },
  open () {
    this.setState({show: true})
  },
  close () {
    this.setState({show: false})
  },
  // functions for adding and deleting notes
  delete (id) {
    this.props.deleteNote(id)
  },
  handleSubmit () {
    const note = {
      flat_id: this.props.flat,
      content: this.props.newNote,
      author: this.props.user.firstName
    }
    this.props.addNote(note)
    this.close()
  },
  render () {
    const {notes} = this.props
    return (
      <div className='Notes'>
        <h1 className='NoteTitle'>Notes</h1>
        <Button className='NoteAdd' onClick={this.open}>Add new</Button>
        <Modal show={this.state.show} onHide={this.close}>
          <NewNoteForm onSubmit={this.handleSubmit} />
        </Modal>

        <Col className='NotesContainer box'>
          {Object.keys(notes).map(noteId => {
            return (
              <div className='NoteFull' key={noteId}>
                <button className='NoteDelete' onClick={() => this.delete(noteId)}>
                  <img src='/images/delete-cross.svg' width='15px' />
                </button>
                <div className='NoteContent'>{notes[noteId].content}</div>
                <div className='NoteAuthor'> - {notes[noteId].author}</div>
              </div>
            )
          })}
        </Col>
      </div>
    )
  }
})
