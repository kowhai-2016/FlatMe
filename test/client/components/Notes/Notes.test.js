/* eslint-env jest */

import React from 'react'
import renderer from 'react-test-renderer'

import Notes from '../../../../src/client/components/Notes'

test('Notes renders correctly', () => {
  const fetchNotes = () => {}
  const notes = []
  const params = {
    id: '1'
  }
  const tree = renderer.create(
    <Notes
      fetchNotes={fetchNotes}
      notes={notes}
      params={params}
      />
  ).toJSON()
  expect(tree).toMatchSnapshot()
})
