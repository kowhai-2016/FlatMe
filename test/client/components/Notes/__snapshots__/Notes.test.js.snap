exports[`test Notes renders correctly 1`] = `
<div
  className="Notes">
  <h1
    className="NoteTitle">
    Notes
  </h1>
  <button
    className="NoteAdd btn btn-default"
    disabled={false}
    onClick={[Function]}
    type="button">
    Add new
  </button>
  <div
    className="NotesContainer box " />
</div>
`;
